### PR TITLE
Adding CoreDNS as an addon

### PR DESCRIPTION
**Pull request by rajansandeep**
Adding CoreDNS as one of the default add-on in minikube. CoreDNS will be disabled by default.

### CODE REVIEW

**Comment by rajansandeep** [comment]
@johnbelamaric @chrisohaver

**Comment by johnbelamaric** [comment]
See https://github.com/kubernetes/features/issues/427

File: deploy/addons/coredns/coreDNS-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      labels:
+        k8s-app: coredns
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}]'
+    spec:
+      serviceAccountName: coredns
+      containers:
+      - name: coredns
+        image: coredns/coredns:latest
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-controller.yaml]: Use `coredns/coredns:011`; we will updated it when we release `1.0.0`.

File: deploy/addons/coredns/coreDNS-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: EnsureExists
+data:
+  Corefile: |
+    .:53 {
+        errors
+        log stdout
+        health
+        kubernetes cluster.local
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-configmap.yaml]: We should have the service CIDR here too for reverse lookups.

File: deploy/addons/coredns/coreDNS-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  name: coredns
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: EnsureExists
+data:
+  Corefile: |
+    .:53 {
+        errors
+        log stdout
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-configmap.yaml]: You should just put `log` not `log stdout`.

File: deploy/addons/coredns/coreDNS-controller.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          successThreshold: 1
+          failureThreshold: 5
+      dnsPolicy: Default
+      volumes:
+        - name: config-volume
+          configMap:
+            name: coredns
+            items:
+            - key: Corefile
+              path: Corefile
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-controller.yaml]: Fix missing newline.

File: deploy/addons/coredns/coreDNS-clusterrole.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- apiGroups:
+  - ""
+  resources:
+  - endpoints
+  - services
+  - pods
+  - namespaces
+  verbs:
+  - list
+  - watch
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-clusterrole.yaml]: Fix missing newline.

File: deploy/addons/coredns/coreDNS-crbinding.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    addonmanager.kubernetes.io/mode: EnsureExists
+  name: system:coredns
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:coredns
+subjects:
+- kind: ServiceAccount
+  name: coredns
+  namespace: kube-system
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-crbinding.yaml]: Fix missing newline.

File: deploy/addons/coredns/coreDNS-sa.yaml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: coredns
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-sa.yaml]: Fix missing newline.

File: deploy/addons/coredns/coreDNS-svc.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+  - name: metrics
+    port: 9153
+    protocol: TCP
```
Comment (johnbelamaric) [inline deploy/addons/coredns/coreDNS-svc.yaml]: newline

**Comment by r2d4** [comment]
@minikube-bot ok to test

**Comment by luxas** [comment]
@rajansandeep Can you now send a new PR to align these manifests with what got merged in kubeadm?
There weren't many changes, but many enough that it's worthwhile to be consistent

**Comment by rajansandeep** [comment]
@luxas Made the changes here. #2179

**Pull request merged**